### ISSUE TITLE
qwen2 系列模型后续是否会支持 fp16 微调？

### ISSUE THREAD

**Issue by XiaoMaGe-hero** [modelscope/ms-swift#1262]
如题

**Comment by liaoduoduo** [comment]
同问

**Comment by 1148270327** [comment]
probability tensor contains either inf, nan or element < 0
这个报错也可能该do_sample=false有效，我是调试deepseek-7b-vl版本，发现这里有问题
 # token selection
            if do_sample:
                probs = nn.functional.softmax(next_token_scores, dim=-1)
                next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
            else:
                next_tokens = torch.argmax(next_token_scores, dim=-1)
